### ISSUE TITLE
Submenus in Preprocess Spectra

### ISSUE THREAD

**Issue by borondics** [Quasars/orange-spectroscopy#446]
More and more method-specific preprocessors will be coming to Orange Spectroscopy and therefore I think it would be good to have the possibility of adding submenus for those to the list of preprocessors in order to keep it structured and easy to read.

**Comment by borondics** [comment]
We would need this very much I think.

Requests for different preprocessors are coming in, for example new baseline correction methods, and simply extending the currently existing preprocessors is not always feasible. 

Even now, we are simply disabling some of the controls for the Rubberband and Linear baseline preprocessors because they only relevant for one of them. Ideally these should be two different GUIs entirely so we either solve to build the preprocessor GUI on the fly depending on which one is selected or make a submenu in the main preprocessor selector for baselines and then we can add new methods easier.

I don' think I can implement this especially for the reason of backward compatibility.

@stuart-cls, @markotoplak thoughts?

**Comment by stuart-cls** [comment]
I agree that the baseline preprocessors should be updated and brought up to our current standards.

It's not too hard to add some dynamic GUI to a preprocessor "Editor" panel: the old Integrate interface does it for example. Taking Normalize as an example: if you think there are too many radio buttons (I guess that is the motivation?) we could change to a combobox and then any required subsettings can be popped in below without too much trouble.

Or maybe what I describe is not what you had in mind :)
